examples/hybrid/staggered_nonhydrostatic_model.jl

```diff
@@ -335,6 +369,13 @@ function Wfact!(W, Y, p, dtγ, t)
     # ᶠw == ᶠw_data .* ᶠw_unit. The Jacobian blocks involve ᶠw_data, not ᶠw.
     ᶠw_data = ᶠw.components.data.:1
 
+    # If ∂(ᶜarg)/∂(ᶠw_data) = 0, then
+    # ∂(ᶠupwind_product(ᶠw, ᶜarg))/∂(ᶠw_data) =
+    #     ᶠupwind_product(ᶠw + εw, arg) / to_scalar(ᶠw + εw).
+    # The εw is only necessary in case w = 0.
+    εw = Ref(Geometry.Covariant3Vector(eps(FT)))
```

Comment:
The value of `ᶠupwind_product(ᶠw, ᶜarg)` is proportional to `ᶠw_data = to_scalar.(ᶠw) = ᶠw.components.data.:1` at every point. So, its derivative is technically just `ᶠupwind_product(ᶠw, ᶜarg) ./ ᶠw_data`. Unfortunately, if `ᶠw_data` is 0, this results in `NaN`s in the Jacobian; since `0 * NaN = NaN`, this in turn results in `NaN`s in the linear solver's output. So, I need the `εw` to avoid that.